docs/create_your_own_brick.rst

```diff
@@ -0,0 +1,387 @@
+Create your own brick
+=====================
+
+This tutorial explains how to create a custom brick, which is useful if you
+want to factorize a specific several combined operations (which can be made of
+bricks themselves) into a single higher-level operation so that you can easily
+reuse it.
+
+The first part of this tutorial lists the requirements and optional components
+that a brick should/can implement while the second part describes the
+construction of a simple toy brick.
+
+This tutorial assumes that you are already familiar with
+:doc:`bricks <bricks_overview>` and how to use them from a user point of view.
+
+
+Bricks ingredients and recipe
+-----------------------------
+
+All the bricks in blocks inherit directly or indirectly from the
+:class:`.Brick`. There is already a rich inheritance hierarchy of
+bricks implemented in blocks and thus, you should consider which brick level
+to inherit from. Bear in mind that multiple inheritance is often possible and
+advocated whenever it makes sense.
+
+Here are examples of possible bricks to inherit from:
+
+* :class:`.Sequence`: a sequence of bricks.
+* :class:`.Initializable`: a brick that defines a same initialiation scheme
+  (weights and biases) for all its children.
+* :class:`.Feedforward`: declares an interface for bricks with one input and
+  one output.
+* :class:`.Linear`: a linear transformation with optional bias. Inherits from
+  :class:`.Initializable` and :class:`.Feedforward`.
+* :class:`.BaseRecurrent`: the base class for recurrent bricks. Check the
+  :doc:`tutorial about rnns</rnn>` for more information.
+* many more!
+
+Let's say that you want to create a brick from scracth, simply inheriting
+from :class:`.Brick`, then you should consider overwriting the following
+methods (strictly speaking, all these methods are optional):
+
+* :meth:`.Brick.__init__`: you should pass by argument the attributes of your
```

Comment:
Here is a question: I do not remember exactly if we have now a documentation that explains the life-cycle of the brick: creation, allocation of parameters, initialization of parameters, application. One can infer them from this document, but I think it must be stated clearly somewhere...